### PR TITLE
Position controller relax fix

### PR DESCRIPTION
**Pull request by lthall**
This PR addresses a problem where high vibrations can exceed the maximum acceleration limit during initialisation and being stored in the I term via the output. This change cleans up the relax functions and calculates them based on their existing value not the output.

Discussion of problem here:
https://discuss.ardupilot.org/t/how-to-turn-off-the-cyclic-blade-pitch-and-collective-pitch-when-the-helicopter-is-on-the-ground/80038/40

### CODE REVIEW

**Comment by bnsgeyer** [comment]
@lthall and @rmackay9  Had a little bit of a scare today when testing this fix.  I initially did a couple of takeoffs and landings in stabilize mode just to see that the relax feature didn't affect the pitch,, roll and yaw axes.  Then I switched to althold and performed a takeoff and landing,  Just after touchdown, the aircraft leaped into the air about 2 ft and then right back down to the ground.  No damage to the heli, but there is something not right here.  Here is a link to the log.  Sad to say but I don't think I have any other time on 4.2-dev.  So I don't have any flights in this firmware version to compare this to.  [Log of flight](https://drive.google.com/file/d/1-AGtDMd8pjwmY5ROKlIeGGNK0ERTwLNl/view?usp=sharing)

I don't recommend any other heli users fly this until we figure this out.

**Comment by bnsgeyer** [comment]
@lthall and @rmackay9 I believe I found the issue.   So to kill two birds with one stone, I cherry picked Leonards commits for this PR into @hendjoshsr71 airspeed PR (#18813) .  His PR branch had more recently been rebased which included this PR (#19648) of Leonard's that was merged last week. Somehow cherry picking the commits from this PR into the more recently rebased Airspeed PR did not throw any merge conflicts.  [Here ](https://github.com/ArduPilot/ardupilot/pull/19648/files#diff-9fe7f1dfc4ef9b28922f55104d879c4b78656bc70293cde53f3c114dad535eeeR765) is the line that was added in the (#19648) PR that caused my issue.  I tried rebasing Leonard's PR branch on the current master and it didn't throw a problem, so be careful rebasing this PR.  I will compile Leonard's PR branch and fly it tomorrow.

**Comment by bnsgeyer** [comment]
@lthall and @rmackay9 Just to be clear, don't blindly merge this PR.  Even though it says there are no conflicts, it will result in the same bug that I found today.

**Comment by bnsgeyer** [comment]
@lthall  Flew your PR branch and everything looked good while flying it.  There was some interesting increase in the collective just after land_complete.  It is only a rise to 0.12 and appears to occur when the controller is relaxed.  I believe it is in part due to the target descent rate going from a commanded descent to 0 climb rate in the relaxed state.  Its something that I don't think is desired.  [Here](https://drive.google.com/file/d/1oZvhD3_NChT3pDIfiDOkpYVJC9xfKluc/view?usp=sharing) is a link to the log and I've included a plot below showing this undesirable increase when the controller is relaxed.

image

**Review by bnsgeyer — APPROVED** [review]
Tested this yesterday on my heli.  No issues.

File: .gitignore [hunk 1]
```diff
@@ -146,3 +146,5 @@ dumpstack_*out
 build.tmp.binaries/
 tasklist.json
 modules/esp_idf
+
```
Comment (tridge) [inline .gitignore:149]: we should not add the ignore of modules/

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by rmackay9 — APPROVED** [review]
LGTM

**Pull request merged**